### ISSUE TITLE
How to retrain slim-MobilenetV2

### ISSUE THREAD

**Issue by Zeitzmz** [JiahuiYu/slimmable_networks#11]
Hi, I'm very curious about you slimmable nas. I retrain standard mobilenetV2 1× and slimmable mobilenetV2 1× for 100 epoch, get both top1:71.8%

您好，我对你的slimmable nas搜索channel特别感兴趣，特别棒的一个工作。我尝试自己去复现训练和搜索、重新训练的过程。训练mobilenet-V2的supernet 50个epoch,　最大width 1.5×在supernet上测试精度65%左右。经过greedy search之后，得到的1*搜索结构和文章图表提供的略有差异，大体一样。然后我重新训练标准的mobilenetV2 1×和搜索出来的mobilenetV2 1× 100个epoch(batchsize:1024 wd:4e-5, lr 4e-1->1e-4(linear decay))，二者top-1都在71.8%左右。

您是怎么重新训练裁剪后的模型呢？可不可以提供一个裁剪后top1 74%的mobilenetV2的pytorch 模型文件呢？十分感谢

**Comment by JiahuiYu** [comment]
Hi @Zeitzmz ,

Thanks for your interest.
I guess you implementation may have bug. On our side, the 1.5x us_mobilenet v2 has around 72% accuracy after 50 epochs, which is almost 7% absolute better than yours. For code release, please refer to issue #7

**Comment by Zeitzmz** [comment]
> Hi @Zeitzmz ,
> 
> Thanks for your interest.
> 
> I guess you may have an implementation issue. On our side, the 1.5x us_mobilenet v2 has around 72% accuracy after 50 epochs, which is almost 7% absolute better than yours. For code release, please refer to issue #7

Thanks for your replay. 
我的训练过程是0.15-1.5分为10份，每次iteration训练选取最大1.5和最小0.15，在从中间随机选取２个，用最大的logits去指导其它3个width模型训练。学习率是从3e-1线性递减到1e-4,，总长50 epoch，batchsize=256，weight decay=4e-5。文章中squeezed learning rate schedule指的是？最后1.5*的测试精度是68%，上面写错了但依旧不及72%。

After search, we train the optimized network architectures for full epochs (300 epochs with linearly decaying learning rate for mobile networks)这个retrain的训练策略可以在哪里找到参考吗？

多谢

**Comment by VectorYoung** [comment]
Hi @Zeitzmz 

I am also trying to reproduce the result. May I have you personal contact(like email or Wechat)? Maybe we can have some discussion. Thanks.

**Comment by zh583007354** [comment]
> Hi @Zeitzmz ,
> Thanks for your interest.
> I guess you may have an implementation issue. On our side, the 1.5x us_mobilenet v2 has around 72% accuracy after 50 epochs, which is almost 7% absolute better than yours. For code release, please refer to issue #7
> 
> Thanks for your replay.
> 我的训练过程是0.15-1.5分为10份，每次iteration训练选取最大1.5和最小0.15，在从中间随机选取２个，用最大的logits去指导其它3个width模型训练。学习率是从3e-1线性递减到1e-4,，总长50 epoch，batchsize=256，weight decay=4e-5。文章中squeezed learning rate schedule指的是？最后1.5*的测试精度是68%，上面写错了但依旧不及72%。
> After search, we train the optimized network architectures for full epochs (300 epochs with linearly decaying learning rate for mobile networks)这个retrain的训练策略可以在哪里找到参考吗？
> 多谢

你在测 1.5x 模型的时候有重新统计一遍1.5x 模型每层BN的均值和方差吗，之前和同学讨论过应该是目前代码里的USBN 并不能用直接用在在AutoSlim 中的Evaluate 过程上，因为目前这种USBN统计的均值和方差仍然相当于是每层宽度固定的BN，实际做随机宽度的NAS时inference 阶段应该重新用训练集（2000张图？按照论文中的说法）统计一下确定了每层宽度的网络模型的每层BN的均值和方差然后再做inference。

另外，我暂时只是讨论出这个问题存在可能影响BN统计量不准，但还没实际实现出代码，还有greedy slimming 部分怎么实现的可以请教一下吗。目前根据论文中的一段话（减去我的理解能力，再减去我的代码能力）我还实现不出来，之前没实现过NAS相关的工作。

**Comment by wlguan** [comment]
@Zeitzmz @JiahuiYu 请问greedy search 部分的代码release了吗？
